resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -30,7 +30,7 @@ import org.apache.spark.scheduler.ExecutorExited
 import org.apache.spark.util.Utils
 
 private[spark] class ExecutorPodsLifecycleManager(
-    conf: SparkConf,
+    val conf: SparkConf,
```

Comment:
is exposing this fine? maybe we can create a new `eventHandlerUnderTest` in the new test below as a workaround

Reply:
@onursatici I think its a small bug fix that can be include dhere, I can update the PR description if that helps. I thought of creating a new one but exposing that value was easy for testing purposes. 
I don't expect that to be a problem unless someone tries to modify SparkConf at the backend when he shouldnt. I dont thinkg it is not a big concern.